### PR TITLE
Replace Projection variant in GenericPredicate with AliasEq

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

**Comment by Veykril** [comment]
The display stuff is mainly stubbed out as I was unsure what to exactly do there

File: crates/hir_ty/src/traits/chalk/mapping.rs [hunk 1]
```diff
@@ -383,19 +373,55 @@ impl ToChalk for ProjectionTy {
         }
     }
 }
+impl ToChalk for OpaqueTy {
+    type Chalk = chalk_ir::OpaqueTy<Interner>;
+
+    fn to_chalk(self, db: &dyn HirDatabase) -> Self::Chalk {
+        chalk_ir::OpaqueTy {
+            opaque_ty_id: self.opaque_ty_id.into(),
```
Comment (flodiebold) [inline crates/hir_ty/src/traits/chalk/mapping.rs:381]: I think these `into`s are redundant? they're already the same type

**Review by flodiebold — APPROVED** [review]

**Comment by Veykril** [comment]
bors r=flodiebold

**Comment by lnicola** [comment]
changelog skip

**Pull request merged**